core/iwasm/aot/aot_runtime.c

```diff
@@ -1079,20 +1117,33 @@ aot_lookup_memory(AOTModuleInstance *module_inst, char const *name)
             return module_inst->export_memories[i].memory;
     return NULL;
 #else
+    (void)name;
     (void)module_inst->export_memories;
-    if (!module_inst->memories)
-        return NULL;
-    return module_inst->memories[0];
+    return aot_get_default_memory(module_inst);
```

Comment:
> the default memory may be not exported

Does it mean we can't export the default memory because of WAMR design limitation or something else? It is a key requirement in component model that there might be an assist instance to provide memory to others.

Reply:
No, here this function looks up memory with name, so the memory found (default memory) should be exported with that name, like the above code when WASM_ENABLE_MULTI_MEMORY is not 0.

Reply:
IIUC, the original version is wrong.

``` c
if (!module_inst->memories)  // -> should be `if (!module_inst->export_memory_count)`
  return NULL;
return module_inst->memories[0];